src/plugins/intel_gna/src/optimizer/gna_pass_manager.cpp

```diff
@@ -2582,6 +2587,116 @@ void TransposeWeightsFromNCHWToNHWCPass::run() {
+    //
+    //    FC                         FC              (Eltwise)
+    //    |                          |               (Const)
+    //  Permute                   Reshape            (Permute)
+    // (optional)           (if exists in Original)
+    //    |                          |
+    //  Reshape                     Any
+    // (optional)  Const
+    //    |       /
+    //  Eltwise(sum)
+    //    |
+    //  Any (e.g. ReLU)
+    //--
+    //
+    // NOTE: This pass is implemented to prevent unnecessary roundtrip to memory (additional layer).
+    // It can be fully removed if corresponding LPT transformation is already implemented.
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "FuseFullyConnectedWithEltwisePass");
+    auto isEltwiseWithSum = [](CNNLayerPtr layer) {
+        return layer != nullptr && LayerInfo(layer).isEltwise() && layer->outData.size() &&
+               layer->params["operation"] == "sum";
+    };
+    for (auto& layer : *pLayers) {
+        if (!LayerInfo(layer).isFullyConnected() || !layer->outData.size()) {
+            continue;
+        }
+        CNNLayerPtr next = nullptr;
+
+        auto& fully_connected = layer;
+
+        next = CNNNetGetNextLayerSkipCertain(fully_connected, 0, 0, DoNotSkip, true).first;
+
+        // If 'next' is a Permute, save the pointer
+        auto permute = next && LayerInfo(next).isPermute() ? next : nullptr;
+
+        // If Permute exists, get the next layer
+        next = permute ? CNNNetGetNextLayerSkipCertain(permute, 0, 0, DoNotSkip, true).first : next;
+
+        // If 'next' is a Reshape, save the pointer
+        auto reshape = next && LayerInfo(next).isReshape() ? next : nullptr;
+
+        // If Reshape exists, get the next layer
+        next = reshape ? CNNNetGetNextLayerSkipCertain(reshape, 0, 0, DoNotSkip, true).first : next;
+
+        // Finally get Eltwise, if meets conditions
+        auto eltwise = isEltwiseWithSum(next) ? next : nullptr;
+
+        // If the layer is not Eltwise or does not perform 'sum' operation, we should skip the pass
+        if (!eltwise) {
+            continue;
+        }
+
+        // Find Const layer attached to Eltwise - needed for getting blob data
+        auto eltwise_const = CNNNetPrevLayerSkipCertain(
+            eltwise,
+            0,
+            [](CNNLayerPtr layer) {
+                return !LayerInfo(layer).isConst();
+            },
+            true);
+
+        if (!eltwise_const) {
+            continue;
+        }
+
+        // Find (any) layer after Eltwise
+        auto any_layer = CNNNetGetNextLayerSkipCertain(eltwise, 0, 0, DoNotSkip, true).first;
+        if (!any_layer) {
+            continue;
+        }
+
+        if (permute) {
+            // Connect FC with Reshape if Permute exists (Permute will be removed)
+            if (!IsOneDimLayer(permute) || !CNNRemoveAndConnect(permute)) {
+                continue;
+            }
+        }
+        if (reshape) {
```

Comment:
What if we have several permutes/reshapes between? I think we should update skip condition and search for eltwise and if it is found, move all layers between FC and eltwise after eltwise.

Reply:
Agree.